### ISSUE TITLE
Turret aiming is now often terribly inaccurate.

### ISSUE THREAD

**Issue by onpon4** [naev/naev#543]
I can't pinpoint exactly where this happened, but it seems that one of the recent changes to the auto-aiming system, or perhaps a combination of them, has completely screwed it up somewhere. I'm not entirely sure when and why it happens, but every once in a while, a series of bullets will shoot in a direction that isn't even close to accurate. It can be as much as a 70 degree difference between where the bullets should be going and where they actually go, in some cases.

**Comment by onpon4** [comment]
Here's one way I've figured out that reproduces it: if you're shooting at a disabled ship from the bottom-right, at least with the Heavy Ion Turret, the shot quite consistently is 90 degrees off, shooting up-right instead of up-left.

**Comment by nenau** [comment]
I think, I have an idea : maybe you aren't using the "only shoot inrange weapons" mode. In that case, when the target is not hittable, the flight time of ammo is INFINITY and thus, weapons are shot in the direction of the velocity of the target.

I know how to fix it (probably) : line 131 in pilot_weapon, add something like : 
if (time == INFINITY) time = 0;
I don't have time to do it right now, I'll try it later (or do it yourself).

Thanks for reporting.

**Comment by onpon4** [comment]
> maybe you aren't using the "only shoot inrange weapons" mode.

This is incorrect.

**Comment by nenau** [comment]
Ok, forget what I wrote in last post, it's already done in aimTurret.

I spotted the behaviour you described, and indeed, it's not supposed to happen. It looks like I made a mistake somewhere and I will try to find it.

About what you wrote in the pull request section : this method doesn't try to predict the behaviour of the ship, it's very close to the previous method (consider that the vectorial velocity will stay the same), but there was an approximation in the computation of the flight time that isn't done in the method I proposed, that's why it should be more accurate (if one gets rid of the bug).
